src/main/java/com/sprint/mission/discodeit/config/AsyncConfig.java

```diff
@@ -0,0 +1,80 @@
+package com.sprint.mission.discodeit.config;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.MDC;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.task.TaskDecorator;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.core.task.support.CompositeTaskDecorator;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+@Configuration
+@EnableAsync
+public class AsyncConfig {
+
+    @Bean(name = "binaryContentTaskExecutor")
+    public TaskExecutor binaryContentTaskExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(5);
+        executor.setMaxPoolSize(10);
+        executor.setQueueCapacity(100);
+        executor.setThreadNamePrefix("BinaryContent-");
+        executor.setTaskDecorator(new CompositeTaskDecorator(
+            List.of(mdcTaskDecorator(), securityContextTaskDecorator())));
+        executor.initialize();
+        return executor;
+    }
+
+    @Bean(name = "eventTaskExecutor")
+    public TaskExecutor eventTaskExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(4);
+        executor.setMaxPoolSize(8);
+        executor.setQueueCapacity(100);
+        executor.setThreadNamePrefix("NotificationEvent-");
+        executor.setTaskDecorator(new CompositeTaskDecorator(
+            List.of(mdcTaskDecorator(), securityContextTaskDecorator())));
+        executor.initialize();
+        return executor;
+    }
+
+    public TaskDecorator mdcTaskDecorator() {
+        return runnable -> {
+            // 현재 스레드 MDC에서 requestId 꺼냄
+            Optional<String> requestId = Optional.ofNullable(
+                MDC.get(MDCLoggingInterceptor.REQUEST_ID));
+            return () -> {
+                // 비동기 스레드에 requestId 설정
+                requestId.ifPresent(id -> MDC.put(MDCLoggingInterceptor.REQUEST_ID, id));
+                try {
+                    // 비동기 작업 실행
+                    runnable.run();
+                } finally {
+                    // 작업 끝나면 clean-up
+                    requestId.ifPresent(id -> MDC.remove(MDCLoggingInterceptor.REQUEST_ID));
+                }
+            };
+        };
+    }
```

Comment:
@kkwan99 부분적으로 부분적 컨텍스트를 복사하면 향후 확장성 측면에서 계속 코드를 추가해야 할 거예요. 새로운 MDC 속성이 추가되면 코드 수정이 필요하니까요. 현재 스레드의 MDC 컨텍스트 전체를 복사하는 방식은 어떨까요?

```suggestion
 public TaskDecorator mdcTaskDecorator() {
    return runnable -> {
        Map<String, String> contextMap = MDC.getCopyOfContextMap();
        return () -> {
            if (contextMap != null) {
                MDC.setContextMap(contextMap);
            }
            try {
                runnable.run();
            } finally {
                MDC.clear(); // 더 확실한 정리
            }
        };
    };
}
```